odl/contrib/fom/supervised.py

```diff
@@ -328,7 +369,7 @@ def range_difference(data, ground_truth, mask=None, normalized=False):
 
 def blurring(data, ground_truth, mask=None, normalized=False,
              smoothness_factor=None):
-    """Return weighted L2 distance, emphasizing regions defined by ``mask``.
+    r"""Return weighted L2 distance, emphasizing regions defined by ``mask``.
```

Comment:
How is this different from the L2 distance above?

Reply:
It looks like it does some kind of smoothing before it takes an L2 distance... I don't know. Maybe we can just add that possibility to the mean squared error function instead?

Reply:
L2 already has a mask, I don't see any difference.

Reply:
The return statement in the function is `mean_squared_error(data, ground_truth, mask, normalized)`, and the only thing it does before is some preprocessing of the mask. So yes, it is obvious that this is the L2 distance.

However, the real question should maybe be *why* someone thought that this kind of preprocessing of the mask was relevant. That I cannot answer this, but I assume that it does some blurring to, e.g., reduce the effect to off-by-one-pixel misalignment in a reconstruction. If someone knows what the idea behind this was, please fill in. @davlars @zickert @ozanoktem @bgris or anyone else how could have been here when it was first implemented.

My suggestion: redo it as a utility function that returns the smoothed mask, which can then be applied in any fom that takes a mask as input. Sounds ok?

Reply:
Regarding why this was implemented at the first "reconstruction pipeline workshop", I think inspiration for some of the figures of merit (including this one) were taken from section 2.4 in this paper: https://www.sciencedirect.com/science/article/pii/S1047847701943382

@aringh's suggestion sounds reasonable to me.